docs/tutorials/basic/data.md

```diff
@@ -192,17 +128,390 @@ class SimpleIter:
+
+
+
+```python
+# unpack_img
+print(mx.recordio.unpack_img(s))
+```
+
+    (HEADER(flag=0, label=1.0, id=0, id2=0), array([[1, 1, 1],
+           [1, 1, 1],
+           [1, 1, 1]], dtype=uint8))
+
+
+
+#### Using tools/im2rec.py
+You can also convert raw images into *RecordIO* format using the ``im2rec.py`` utility script that is provided in the *MXNet* [src/tools]() folder.  
+An example how to use the script for converting to *RecordIO* format is shown in the `Image IO` section below.
+
+
+## Image IO
+
+In this section we will learn how to preprocess and load image data in *MXNet*.
+
+There are 4 ways of loading image data in *MXNet*.
+   1. Using [__mx.image.imdecode__](http://mxnet.io/api/python/io.html#mxnet.image.imdecode) to load raw image files.
+   2. Using [__`mx.img.ImageIter`__](http://mxnet.io/api/python/io.html#mxnet.image.ImageIter) implemented in ``Python`` which is very flexible to customization. It can read from .rec(`RecordIO`) files and raw image files.
+   3. Using [__`mx.io.ImageRecordIter`__](http://mxnet.io/api/python/io.html#mxnet.io.ImageRecordIter) implemented on the MXNet backend in C++. This is less flexible to customization but provides various language bindings.  
+   4. Creating a Custom Iterator inheriting `mx.io.DataIter`
+
+
+First set the environment variable `MXNET_HOME` to the root of the *MXNet* source folder
+
+
+```python
+# change this to your mxnet location
+MXNET_HOME = '/scratch/mxnet'
+```
+
+
+### Preprocessing Images
+Images can be preprocessed in may different ways, some of them are listed below:
+- Using `mx.io.ImageRecordIter` which is fast but not very flexible. It is great for simple tasks like image recognition but won't work for more complex tasks like detection and segmentation.
+- Using `mx.recordio.unpack_img` (or `cv2.imread`, `skimage`, etc) + `numpy` is flexible but slow due to ``Python`` Glolbal Interpreter Lock(GIL).  
+- Using *MXNet* provided `mx.image` package. It stores images in [__`NDArray`__](http://mxnet.io/tutorials/basic/ndarray.html) format and leverages MXNet's [dependency engine](http://mxnet.io/architecture/note_engine.html) to automatically parallelize processing and circumvent GIL.
+
+We will show below some of the frequently used preprocessing routines provided by the `mx.image` package.
+
+Let's download sample images that we can work with.
+
+
+```python
+os.system('wget http://data.mxnet.io/data/test_images.tar.gz')
+os.system('tar -xf test_images.tar.gz')
+```
+
+
+
+
+#### Loading raw images
+Using `mx.image.imdecode` let us first load the images, `imdecode` provides a similar interface to ``OpenCV``.  
+**Note: ** You will still need ``OpenCV``(not the CV2 Python library) installed to use `mx.image.imdecode`.
+
+
+```python
+import cv2
+import time
+tic = time.time()
+N = 1000
+for i in range(N):
+    img = mx.image.imdecode(open('test_images/ILSVRC2012_val_00000001.JPEG').read())
```

Comment:
Loading the same image again and again doesn't look nice. Can we decode 1000 different images?